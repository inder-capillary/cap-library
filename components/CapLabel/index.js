/**
*
* CapLabel
*
*/

import styled from 'styled-components';
import PropTypes from 'prop-types';

const labels = {
  label1: {
    'color': '#5e6c84',
    'font-size': '12px',
    'font-weight': 'normal',
    'line-height': '20px',
  },
  label2: {
    'color': '#091e42',
    'font-size': '12px',
    'font-weight': 'normal',
    'line-height': '20px',
  },
  label3: {
    'color': '#97a0af',
    'font-size': '12px',
    'font-weight': 'normal',
    'line-height': '20px',
  },
};

const CapLabel = styled.div`
    font-size: ${[(props) => labels[props.type]["font-size"]]};
    font-weight: ${[(props) => labels[props.type]["font-weight"]]};
    color: ${[(props) => labels[props.type].color]};
    line-height: ${[(props) => labels[props.type]["line-height"] || 'initial']};
    `;

const CapLabelInline = styled.span`
font-size: ${[(props) => labels[props.type]["font-size"]]};
font-weight: ${[(props) => labels[props.type]["font-weight"]]};
color: ${[(props) => labels[props.type].color]};
line-height: ${[(props) => labels[props.type]["line-height"] || 'initial']};
`;

CapLabel.propTypes = {
  type: PropTypes.string,
  children: PropTypes.any,
};

CapLabel.CapLabelInline = CapLabelInline;

CapLabel.defaultProps = {
  type: 'label1',
};
export default CapLabel;
