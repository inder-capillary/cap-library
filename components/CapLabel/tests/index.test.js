import React from 'react';
import renderer from 'react-test-renderer';
import CapLabel from '../index';

describe('<CapLabel />', () => {
  it('CapLabel renders correctly', () => {
    const tree = renderer.create(<CapLabel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
