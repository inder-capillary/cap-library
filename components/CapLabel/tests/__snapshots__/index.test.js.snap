// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapLabel /> CapLabel renders correctly 1`] = `
<div
  className="sc-bdVaJa fZdJKj"
  type="label1"
/>
`;
